src/components/uncorrelatedEvents.js

```diff
@@ -0,0 +1,94 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import { connect } from 'react-redux'
+import EventFilter from 'components/Timeline/EventFilter'
+import Footer from 'components/Timeline/EventFooter'
+import Events from 'components/Timeline/Events'
+import { DateTime } from 'luxon'
+import * as filterActions from 'actions/filterActions'
+import TimePicker from 'material-ui/TimePicker'
+import DatePicker from 'material-ui/DatePicker'
+import { FlatButton } from 'material-ui'
+
+class uncorrelatedEvents extends Component {
+  static propTypes = {
+    events: PropTypes.object.isRequired,
+    filters: PropTypes.object,
+    eventTypes: PropTypes.object.isRequired,
+    dispatch: PropTypes.func.isRequired
+  }
+
+  componentDidMount () {
+    const { filters, dispatch } = this.props
+    const newFilter = Object.assign({}, filters, {incidentId: null})
+    dispatch(filterActions.clearFilterIncidentId(newFilter))
+  }
+  constructor (props) {
+    super(props)
+    this.state = {startDate: null, startTime: null, endDate: null, endTime: null}
+  }
+
+  render () {
+    const { events, dispatch, eventTypes, history } = this.props
+    return (
+      <div >
+        <DatePicker
```

Comment:
This filtering interface looks quite useful, to the point where I would prefer to include it on the main Timeline component as well. Extracting this to its own component would allow re-use and also vastly simplify the uncorrelatedEvents component by extracting all the stuff that deals with React local state.

That said, this is another place where I think it makes a lot of sense to break the filter Reducer into multiple combined reducers; these pickers could dispatch actions to update just their portion of the Redux state. Not sure if that's worth the extra time investment for now, though.